### PR TITLE
test: fix user without process privilege can access cluster_tidb_trx

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/26127

### What is changed and how it works?

*: What's Changed: add the privileges check when read the memory table

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- test: fix user without process privilege can access cluster_tidb_trx

### CODE REVIEW

File: infoschema/tables_test.go [hunk 1]
```diff
@@ -798,6 +798,28 @@ func (s *testClusterTableSuite) TestForClusterServerInfo(c *C) {
 		c.Assert(gotAddrs, DeepEquals, cas.addrs, Commentf("sql: %s", cas.sql))
 		c.Assert(gotNames, DeepEquals, cas.names, Commentf("sql: %s", cas.sql))
 	}
+
+	// More test about the systemInfo's privileges.
+	tk.MustExec("create user 'testuser'@'localhost'")
```
Comment (mjonss) [inline infoschema/tables_test.go:803]: Could you also add a
```
defer tk.MustExec("drop user 'testuser'@'localhost'")
```
?

File: infoschema/tables_test.go [hunk 2]
```diff
@@ -867,6 +889,34 @@ func (s *testClusterTableSuite) TestSelectClusterTable(c *C) {
 	}
 }
 
+func (s *testClusterTableSuite) TestClusterTablePrivileges(c *C) {
+	tk := s.newTestKitWithRoot(c)
+	// More tests about the Cluster_TiDB_Trx's privileges.
+	tk.MustExec("create user 'testuser'@'localhost'")
```
Comment (mjonss) [inline infoschema/tables_test.go:895]: Could you also add a
```
defer tk.MustExec("drop user 'testuser'@'localhost'")
```
?

File: infoschema/tables_test.go [hunk 3]
```diff
@@ -867,6 +889,34 @@ func (s *testClusterTableSuite) TestSelectClusterTable(c *C) {
...
+	tk.MustExec("create user 'testuser2'@'localhost'")
```
Comment (mjonss) [inline infoschema/tables_test.go:896]: Could you also add a
```
defer tk.MustExec("drop user 'testuser2'@'localhost'")
```
?

File: infoschema/tables_test.go [hunk 4]
```diff
@@ -867,6 +889,34 @@ func (s *testClusterTableSuite) TestSelectClusterTable(c *C) {
...
+	tk.MustExec("grant process on *.* to 'testuser2'@'localhost'")
+	tk.MustExec("flush privileges")
```
Comment (mjonss) [inline infoschema/tables_test.go:898]: Is this needed? I think it only should be needed when manipulating the mysql.* tables directly, not when using grant statements.

**Review by mjonss — COMMENTED** [review]
I think you also fixed:
CLUSTER_SYSTEMINFO, CLUSTER_HARDWARE and CLUSTER_LOAD, or is that just from a previous patch?

File: executor/memtable_reader.go [hunk 5]
```diff
@@ -293,6 +293,10 @@ type clusterServerInfoRetriever struct {
 
 // retrieve implements the memTableRetriever interface
 func (e *clusterServerInfoRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	// To access TableClusterSystemInfo, TableClusterHardware, TableClusterLoad, PROCESS privileges is needed.
```
Comment (djshow832) [inline executor/memtable_reader.go:296]: This is another problem. Does this PR forbid 4 tables?

File: infoschema/tables_test.go (hunk 1 above)
Comment (AilinKid) [reply infoschema/tables_test.go:803]: sure~ thx

File: infoschema/tables_test.go (hunk 4 above)
Comment (AilinKid) [reply infoschema/tables_test.go:898]: I see, got it

**Comment by MyonKeminta** [comment]
In our design `TIDB_TRX` and `CLUSTER_TIDB_TRX` shows all information to users with PROCESS privilege, but it only shows transactions executed by the current user if the user doesn't have PROCESS privilege. Is this design improper or you misunderstood its behavior?

**Comment by AilinKid** [comment]
> In our design `TIDB_TRX` and `CLUSTER_TIDB_TRX` shows all information to users with PROCESS privilege, but it only shows transactions executed by the current user if the user doesn't have PROCESS privilege. Is this design improper or you misunderstood its behavior?

Misunderstood before.

**Comment by MyonKeminta** [comment]
This actually referenced the [processlist](https://dev.mysql.com/doc/refman/5.7/en/show-processlist.html) table:

> If you have the PROCESS privilege, you can see all threads, even those belonging to other users. Otherwise (without the PROCESS privilege), nonanonymous users have access to information about their own threads but not threads for other users, and anonymous users have no access to thread information.

It's a problem that this behavior seems to lack of test coverage.  cc @longfangsong 
And I just notice that the privilege requirements to the system tables are not well explained in our documentation.

**Pull request closed without merging**